servers/rendering/renderer_canvas_cull.cpp

```diff
@@ -2165,6 +2165,30 @@ bool RendererCanvasCull::free(RID p_rid) {
 	return true;
 }
 
+template <class T>
+void RendererCanvasCull::_free_rids(T &p_owner, const char *p_type) {
+	List<RID> owned;
+	p_owner.get_owned_list(&owned);
+	if (owned.size()) {
+		if (owned.size() == 1) {
+			WARN_PRINT(vformat("1 RID of type \"%s\" was leaked.", p_type));
+		} else {
+			WARN_PRINT(vformat("%d RIDs of type \"%s\" were leaked.", owned.size(), p_type));
+		}
+		for (const RID &E : owned) {
+			free(E);
```

Comment:
We don't do this consistantly on the 3D side but if you search you'll see a few places I added it, but seeing that it can be very difficult to find the cause of the leak simply by a count, I check if a description was recorded for the RID (this is a debug/tools build only feature), than we print the description of the resource being leaked.

This also means off course that we need to get more consistent in recording descriptions when the resource is allocated. Something we can improve on over time.